src/Language/Pirouette/PlutusIR/SMT.hs

```diff
@@ -1,76 +1,89 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternSynonyms #-}
+
 module Language.Pirouette.PlutusIR.SMT where
 
-import qualified Data.Text as Text
+import qualified Data.ByteString as BS
+import Data.Text (Text)
 import Language.Pirouette.PlutusIR.Syntax
+import Pirouette.Monad
+import Pirouette.Symbolic.Eval.Helpers
 import Pirouette.Symbolic.Eval.Types
 import Pirouette.SMT.Base
 import Pirouette.SMT.Constraints
+import Pirouette.Term.Syntax.Base
+import Pirouette.Term.Syntax.SystemF as SystemF
 import qualified PlutusCore as P
 import qualified PureSMT
 
+-- See https://github.com/input-output-hk/plutus/blob/master/plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs
+
 instance LanguageSMT PlutusIR where
   translateBuiltinType = trPIRType
 
   -- translateBuiltinTerm = error "translateBuiltinTerm (t :: BuiltinTerms PlutusIR): not yet impl"
   translateBuiltinTerm = trPIRFun
   translateConstant = trPIRConstant
-  isStuckBuiltin = error "isStuckBuiltin (t :: TermMeta PlutusIR meta): not yet impl"
 
-instance LanguageSymEval PlutusIR where
-  branchesBuiltinTerm _tm _translator _args = pure Nothing
+  isStuckBuiltin e
+    | termIsConstant e = True
+    | Just _ <- termIsMeta e = True
+  isStuckBuiltin (App (Free (Builtin op)) args)
+    | op `elem` plutusIRBasicOps || op `elem` plutusIRBasicRels
+    = let args' = map (\(TermArg a) -> a) args
```

Comment:
It's guaranteed if the term is well-typed (since those operations take no type arguments). But you are right that including a safeguard won't hurt either...